core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/AffectedEntitiesUpdater.java

```diff
@@ -122,21 +89,19 @@ private boolean updateShadowVariable(EntityVariablePair<Solution_> entityVariabl
         if (loopDescriptor != null) {
             var oldLooped = (boolean) loopDescriptor.getValue(entity);
             if (oldLooped != isLooped) {
-                // Loop status change; add to affected entities
-                affectedEntities.add(entityVariable);
+                // Loop status change; mark it
+                changeShadowVariableAndNotify(loopDescriptor, entity, isLooped);
             }
         }
 
         if (isLooped) {
             if (oldValue != null) {
-                affectedEntities.add(entityVariable);
```

Comment:
This is effectively dead code. The `add` method would check if the loop descriptor is not null and only scheduled the update in that case. Since the update in this case already happened above, this code is no longer necessary.